File: day3/Exercise_day3/supplementary_material_mpi/allreduce.f90
!----------------------------------------------------------------------                                                               
!  a grogram to demonstrate a reduce to all                              
!----------------------------------------------------------------------   

      program allreduce
      
      use mpi

      implicit none

      integer :: ierror, my_rank
      integer :: sum

!----------------------------------------------------------------------         
      call MPI_Init(ierror)
      call MPI_Comm_rank(MPI_COMM_WORLD, my_rank, ierror)

!.....TBD calculate sum of all ranks
      write(*,*) "Rank", my_rank, ": Sum =", sum

      call MPI_Finalize(ierror)

!----------------------------------------------------------------------         
      end program allreduce
!----------------------------------------------------------------------   
      

File: day3/Exercise_day3/supplementary_material_mpi/allreduce.cpp
#include <stdio.h>
#include <mpi.h>


int main (int argc, char *argv[])
{
    int my_rank, size;
    int sum;


    MPI_Init(&argc, &argv);

    MPI_Comm_rank(MPI_COMM_WORLD, &my_rank);

    MPI_Comm_size(MPI_COMM_WORLD, &size);

    /* Compute sum of all ranks. */

    printf ("Rank %i:\tSum = %i\n", my_rank, sum);

    MPI_Finalize();
    return 0;
}
